components/patient-data/ui/src/main/resources/PhenoTips/JSONImportDialog.xml

```diff
@@ -173,24 +178,36 @@
       });
 
       this.textareaChangeFunction = function() {
-        importButton.disabled = _this.jsonData.value.trim() == "";
+        importButton.disabled = _this.jsonData.value.strip() == "";
+      };
+
+      this.textareaTextEntered = function() {
+        if (_this.jsonData.value.strip() != "")
+          _this._processImportedData();
       };
 
       // different browsers require different event for onchange detection (special cases: IE, paste, backspace)
       var eventList = ["input","propertychange", "mousemove", "keyup"];
       eventList.each(function(item) {
         _this.jsonData.observe(item, _this.textareaChangeFunction);
       });
+
+      if (this._builder.performIdCheck)
```

Comment:
Always use `{` and `}`, even around a single line, since it prevents bugs when adding new lines in the future.